WooCommerce/Classes/ViewRelated/Reviews/ReviewReply.swift

```diff
@@ -1,21 +1,71 @@
 import Foundation
 import SwiftUI
+import Combine
 
 /// Hosting controller that wraps a `ReviewDetailsReply` view.
 ///
 final class ReviewReplyHostingController: UIHostingController<ReviewReply>, UIAdaptivePresentationControllerDelegate {
 
     private let viewModel: ReviewReplyViewModel
 
-    init(viewModel: ReviewReplyViewModel) {
+    /// Presents notices in the tab bar context.
+    ///
+    private let systemNoticePresenter: NoticePresenter
+
+    /// Presents notices in the current modal presentation context.
+    ///
+    private lazy var modalNoticePresenter: NoticePresenter = {
+        let presenter = DefaultNoticePresenter()
+        presenter.presentingViewController = self
+        return presenter
+    }()
+
+    /// Emits the intent to present a `ReviewReplyNotice`
+    ///
+    private lazy var presentNoticePublisher = {
+        viewModel.presentNoticeSubject.eraseToAnyPublisher()
+    }()
+
+    /// References to keep the Combine subscriptions alive within the lifecycle of the object.
+    ///
+    private var subscriptions: Set<AnyCancellable> = []
+
+    init(viewModel: ReviewReplyViewModel,
+         noticePresenter: NoticePresenter = ServiceLocator.noticePresenter) {
         self.viewModel = viewModel
+        self.systemNoticePresenter = noticePresenter
         super.init(rootView: ReviewReply(viewModel: viewModel))
 
         // Needed because a `SwiftUI` cannot be dismissed when being presented by a UIHostingController
         rootView.dismiss = { [weak self] in
             self?.dismiss(animated: true, completion: nil)
         }
 
+        // Observe the present notice intent.
+        presentNoticePublisher
+            .compactMap { $0 }
+            .sink { [weak self] notice in
+                switch notice {
+                case .success:
+                    self?.systemNoticePresenter.enqueue(notice: Notice(title: Localization.success, feedbackType: .success))
+                case .error:
+                    let noticeIdentifier = UUID().uuidString
```

Comment:
Is this to allow multiple error notices to be displayed?

If so: 
1. Neat trick!
2. We should probably add a comment explaining it! 😅

Reply:
This is so we can provide `NoticeNotificationInfo` to the `Notice`, so the notice will be presented as a system notification when the app isn't in the foreground:

https://github.com/woocommerce/woocommerce-ios/blob/81ed7b18078c58d3337f53a91d0f09c802bf3c3f/WooCommerce/Classes/Tools/Notices/Notice.swift#L26-L28

I borrowed the idea from our `EditCustomerNote` implementation, with the idea that this way if someone replies to a review and then closes the app, they'll still get a notification if the reply fails.

I'll add a comment to explain!